kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/eventually.kt

```diff
@@ -205,7 +211,7 @@ object NoopEventuallyListener : EventuallyListener {
 private class EventuallyControl(val config: EventuallyConfiguration) {
 
    val start = timeInMillis()
-   val end = start + config.duration.inWholeMilliseconds
+   val end = addSafely()
```

Comment:
Have you tried to run this? The local `start` variable is not visible in `addSafely()`, right?

What about expressing more clearly what happens, so that readers don't have to look up the implementation of `addSafely()`? For example, mirroring the `coerce...` functions in the stdilb, we could use:

```kotlin
val end = (start + config.duration.inWholeMilliseconds).orInfinityIfNegative()
```
with:
```kotlin
fun Long.orInfinityIfNegative() = if (this < 0) Duration.INFINITE.inWholeMilliseconds else this
```

Reply:
thanks to your advice I refactored code! thanks a lot!